src/evmIndexer/utils/index.ts

```diff
@@ -0,0 +1,338 @@
+/*
+The Licensed Work is (c) 2024 Sygma
+SPDX-License-Identifier: LGPL-3.0-only
+*/
+import { randomUUID } from "crypto";
+import { assertNotNull } from "@subsquid/evm-processor";
+import {
+  AbiCoder,
+  BigNumberish,
+  BytesLike,
+  Contract,
+  Provider,
+  ethers,
+  formatUnits,
+  getBytes,
+} from "ethers";
+import { MultiLocation } from "@polkadot/types/interfaces";
+import { ApiPromise } from "@polkadot/api";
+import { Network, ResourceType } from "@buildwithsygma/sygma-sdk-core";
+import ERC20Contract from "@openzeppelin/contracts/build/contracts/ERC20.json";
+import * as bridge from "../../abi/bridge";
+import { Context, Log } from "../../evmProcessor";
+import {
+  ContractType,
+  DecodedDepositLog,
+  DecodedFailedHandlerExecution,
+  DecodedProposalExecutionLog,
+  DepositType,
+  FeeData,
+} from "../evmTypes";
+import { Domain } from "../../config";
+import { logger } from "../../utils/logger";
+import { Transfer } from "../../model";
+import * as FeeHandlerRouter from "../../abi/FeeHandlerRouter.json";
+
+export const nativeTokenAddress = "0x0000000000000000000000000000000000000000";
+const STATIC_FEE_DATA = "0x00";
+type Junction = {
+  accountId32?: {
+    id: string;
+  };
+};
+type FeeDataResponse = {
+  fee: string;
+  tokenAddress: string;
+};
+
+export async function parseDeposit(
+  log: Log,
+  fromDomain: Domain,
+  toDomain: Domain,
+  provider: Provider,
+  substrateRpcUrlConfig: Map<number, ApiPromise>
+): Promise<DecodedDepositLog> {
+  const event = bridge.events.Deposit.decode(log);
+  const resource = fromDomain.resources.find(
+    (resource) => resource.resourceId == event.resourceID
+  );
+  if (!resource) {
+    throw new Error(
+      `Resource with ID ${event.resourceID} not found in shared configuration`
+    );
+  }
+  const resourceType = resource.type || "";
+  const resourceDecimals = resource.decimals || 18;
+
+  const transaction = assertNotNull(log.transaction, "Missing transaction");
+
+  return {
+    id: log.id,
+    blockNumber: log.block.height,
+    depositNonce: event.depositNonce,
+    toDomainID: event.destinationDomainID,
+    sender: transaction.from,
+    destination: parseDestination(
+      event.data as BytesLike,
```

Comment:
```suggestion
      event.data,
```
this cast makes no sense